packages/storybook/scripts/build.mjs

```diff
@@ -10,7 +10,9 @@ async function main() {
     console.log('Skipping storybook build for stable build');
     return;
   }
-  await exec(quote(['npx', 'build-storybook', '--quiet', '--loglevel', 'warn', ...process.argv.slice(2)]));
+  await exec(quote(['npx', 'build-storybook', '--quiet', '--loglevel', 'warn', ...process.argv.slice(2)]), {
+    NODE_OPTIONS: '--openssl-legacy-provider' // Storybook needs this for use with node 18; resolved in storybook v7
```

Comment:
Will this break if someone builds storybook locally with node 16?

Reply:
Yes, I think so.

Reply:
Yes, this will break for node <18.

Reply:
Hmm, instead of passing the option, can we have this script set an environment variable?
https://stackoverflow.com/a/76178571/3047190

Reply:
Or have the script check the node version and only set this if node 18+

Reply:
Ha, I just implemented this!